### PR TITLE
Udp protocol

### PR DESCRIPTION
**Pull request by tkus1**
Implement UDP server and client.
Also, some useful scripts for setting up the project and doing performance tests are added.

### CODE REVIEW

File: GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public void run() {
+        try {
+            String data = new String(packet.getData(), 0, packet.getLength());
+
+            // 3. Parse operation and key-value
+            String[] parts = data.split(" ");
+            String operation = parts[0];
+            String key = parts.length > 1 ? parts[1] : null;
+            String value = parts.length > 2 ? parts[2] : null;
```
Comment (agamjotpanesar) [inline GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java]: we can extract a new function out of it, just for clean code purposes, that way you wont have to write a comment. The function name itself should be self explanatory

File: GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                server.del(key);
+            } else if (operation.equals("store")) {
+                ArrayList<String> entries = server.store();
+                for(String entry : entries) {
+                    sendResponse(entry);
+                }
+                sendResponse("Finished");
+            } else {
+                server.exit();
+            }
```
Comment (agamjotpanesar) [inline GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java]: here also, extract a new function out of these line for clean code

File: GenericNode/src/main/java/genericnode/Servers/UDPServer.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+
+public class UDPServer implements Server{
+    private DatagramSocket udpServer = null;
+    @Override
```
Comment (agamjotpanesar) [inline GenericNode/src/main/java/genericnode/Servers/UDPServer.java:15]: insert new line here

File: GenericNode/src/main/java/genericnode/Servers/UDPServer.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import javax.xml.crypto.Data;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+
+public class UDPServer implements Server{
+    private DatagramSocket udpServer = null;
```
Comment (agamjotpanesar) [inline GenericNode/src/main/java/genericnode/Servers/UDPServer.java:13]: insert new line above this

File: GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    private void sendResponse(String response) throws IOException {
+        byte[] responseData = response.getBytes();
+        DatagramPacket responsePacket = new DatagramPacket(
+                responseData, responseData.length, packet.getAddress(), packet.getPort());
+        socket.send(responsePacket);
+    }
+    private String receivePacketFromClient() throws IOException {
```
Comment (agamjotpanesar) [inline GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java:47]: always insert new lines between two functions

**Comment by tkus1** [comment]
@agamjotpanesar I modified the codes following your comments. Review it again.

**Pull request merged**